portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -47,23 +47,30 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     if (userService.isUserLoggedIn()) {
-      String userEmail = userService.getCurrentUser().getEmail();
-      String logoutUrl = userService.createLogoutURL(REDIRECT_URL_AFTER_LOGOUT);
-      saveComment(request.getParameter("display-name"), request.getParameter("comment"));
+      // TODO: Use nickname instead
+      saveComment(userService.getCurrentUser().getEmail(), request.getParameter("comment"));
       response.sendRedirect("/aboutme.html");
-    } else {
-      String loginUrl = userService.createLoginURL(REDIRECT_URL_AFTER_LOGIN);
-      response.sendRedirect(loginUrl);
     }
-
   }
 
   private String convertCommentsToJson() {
     Query query = new Query("Comment");
     PreparedQuery results = datastore.prepare(query);
+
     int loggedIn = userService.isUserLoggedIn() ? 1 : 0;
+    String json = "{ \"loggedInStatus\": " + loggedIn;
 
-    String json = "{ \"loggedInStatus\": " + loggedIn +", \"comments\": [";
+    if (userService.isUserLoggedIn()) {
+      String userEmail = userService.getCurrentUser().getEmail();
+      json += ", \"userEmail\": \"" + userEmail + "\"";
+      // FIX: Logout does not work.
+      String logoutUrl = userService.createLogoutURL(REDIRECT_URL_AFTER_LOGOUT);
+      json += ", \"logoutUrl\": \"" + logoutUrl + "\"";
+    } else { 
+      String loginUrl = userService.createLoginURL(REDIRECT_URL_AFTER_LOGIN);
+      json += ", \"loginUrl\": \"" + loginUrl + "\"";
+    }
+    json += ", \"comments\": [";
```

Comment:
I think you should fetch the login status from a different servelet and then fetch the comments if the user is logged in. Otherwise you are passing back all the data even if the user is not logged in.

Reply:
Done: 665a4ce